.github/workflows/2025-4.yml

```diff
@@ -0,0 +1,71 @@
+name: 2025-4 Nightly Build
+
+on:
+  schedule:
+    - cron: '0 2 * * *'
+  workflow_dispatch: ~
+
+jobs:
+  increment-version:
+    runs-on: ubuntu-latest
+    outputs:
+      has_recent_commits: ${{ steps.check-commits.outputs.has_recent_commits }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Configure Git
+        run: |
+          git config --local user.email "action@github.com"
+          git config --local user.name "GitHub Action"
+
+      - name: Check for recent commits
+        id: check-commits
+        run: |
+          # Check if there are commits in the last 24 hours
+          recent_commits=$(git log --since="24 hours ago" --no-merges --oneline | wc -l)
```

Comment:
does this need logic to avoid the previous day's "increment patch" commit?

Reply:
yeah, it specifically looks for `version_commits=$(git log --since="24 hours ago" --grep="Increment patch version" --oneline | wc -l)` which get removed from the list of commits it considers